.github/scripts/validate_components.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+"""Validate that all components and pipelines compile successfully."""
+
+import ast
+import sys
+from pathlib import Path
+
+
+def find_decorated_functions(file_path: Path) -> dict[str, list[str]]:
+    """Find functions decorated with KFP decorators in a Python file.
+    
+    Returns a dict mapping decorator type to list of function names.
+    """
+    try:
+        content = file_path.read_text()
+        tree = ast.parse(content)
+    except (SyntaxError, UnicodeDecodeError) as e:
+        print(f"  Warning: Could not parse {file_path}: {e}")
+        return {}
+
+    component_decorators = {"component", "container_component"}
+    pipeline_decorators = {"pipeline"}
+
+    result = {"components": [], "pipelines": []}
+
+    for node in ast.walk(tree):
+        if isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef)):
+            for decorator in node.decorator_list:
+                decorator_name = None
+                if isinstance(decorator, ast.Name):
+                    decorator_name = decorator.id
+                elif isinstance(decorator, ast.Attribute):
+                    decorator_name = decorator.attr
+                elif isinstance(decorator, ast.Call):
+                    if isinstance(decorator.func, ast.Name):
+                        decorator_name = decorator.func.id
+                    elif isinstance(decorator.func, ast.Attribute):
+                        decorator_name = decorator.func.attr
+
+                if decorator_name in component_decorators:
+                    result["components"].append(node.name)
+                    break
+                elif decorator_name in pipeline_decorators:
+                    result["pipelines"].append(node.name)
+                    break
+
+    return result
+
+
+def validate_imports() -> bool:
+    """Validate that package structure imports correctly."""
+    print("Validating package imports...")
+    success = True
+
+    packages = [
+        ("components", ["training", "evaluation", "data_processing", "deployment"]),
+        ("pipelines", ["training", "evaluation", "data_processing", "deployment"]),
```

Comment:
I don't think these four submodules/categories are guaranteed to be the only ones once contributors start adding thier components - probably need to dynamically fetch them